x-pack/legacy/plugins/lens/public/indexpattern_plugin/field_item.tsx

```diff
@@ -41,22 +97,286 @@ export function FieldItem({ field, indexPattern, highlight, exists }: FieldItemP
+              onClick={() => {
+                if (exists) {
+                  setOpen(!infoIsOpen);
+                  fetchData();
+                }
+              }}
+              onKeyPress={event => {
+                if (exists && event.key === 'ENTER') {
+                  setOpen(!infoIsOpen);
+                  fetchData();
+                }
+              }}
+              title={i18n.translate('xpack.lens.indexPattern.fieldStatsButton', {
+                defaultMessage: 'Click or Enter for more information about {fieldName}',
+                values: { fieldName: field.name },
+              })}
+              aria-label={i18n.translate('xpack.lens.indexPattern.fieldStatsButton', {
+                defaultMessage: 'Click or Enter for more information about {fieldName}',
+                values: { fieldName: field.name },
+              })}
+            >
+              <FieldIcon type={field.type as DataType} />
+
+              <span className="lnsFieldListPanel__fieldName" title={field.name}>
+                {wrappableHighlightableFieldName}
+              </span>
+            </div>
+          </EuiKeyboardAccessible>
+        </DragDrop>
+      }
+      isOpen={infoIsOpen}
+      closePopover={() => setOpen(false)}
+      anchorPosition="rightUp"
+      panelClassName="lnsFieldListPanel__fieldPopover"
     >
-      <div className="lnsFieldListPanel__fieldInfo">
-        <FieldIcon type={field.type as DataType} />
+      <div>
+        {state.isLoading && <EuiLoadingSpinner />}
+
+        {state.histogram && state.topValues && (
+          <>
+            <EuiButtonGroup
+              buttonSize="s"
+              legend={i18n.translate('xpack.lens.indexPattern.fieldStatsDisplayToggle', {
+                defaultMessage: 'Toggle either the',
+              })}
+              options={[
+                {
+                  label: i18n.translate('xpack.lens.indexPattern.fieldTopValuesLabel', {
+                    defaultMessage: 'Top Values',
+                  }),
+                  id: 'topValues',
+                },
+                {
+                  label: i18n.translate('xpack.lens.indexPattern.fieldHistogramLabel', {
+                    defaultMessage: 'Histogram',
+                  }),
+                  id: 'histogram',
+                },
+              ]}
+              onChange={optionId => {
+                setShowingHistogram(optionId === 'histogram');
+              }}
+              idSelected={showingHistogram ? 'histogram' : 'topValues'}
+            />
+            <EuiSpacer />
+          </>
+        )}
+
+        {state.histogram && (!state.topValues || showingHistogram) && (
+          <Chart className="lnsDistributionChart" data-test-subj="lnsFieldListPanel-histogram">
+            <Settings rotation={90} />
+
+            <Axis
+              id={getAxisId('key')}
+              position={Position.Left}
+              tickFormat={d => formatter.convert(d)}
```

Comment:
**Optional**: The default date format is suboptimal for most cases - we could use
```ts
{
  niceTimeFormatter,
} from '@elastic/charts';
```

in this case